Add an autostart object

t-wissmann opened the pull request:
This adds an autostart object which allows modifying the custom
autostart path at run time and also helps querying information about the
last invoked autostart run (pid, status, running).

This closes #762.

t-wissmann commented:
Maybe `last_status` can be simplified to `status` or `exit_status`.

blarz commented on src/autostart.cpp:
```diff
@@ -0,0 +1,90 @@
+                "or $HOME is used.");
+    global_autostart_path_.setDoc("Path of the system-wide autostart, used as a fallback.");
+    pid_.setDoc(
+                "the process id of the last autostart invokation. "
```
invocation

blarz commented on src/main.cpp:
```diff
@@ -464,8 +420,9 @@ int main(int argc, char* argv[]) {
-    // remove zombies on SIGCHLD
-    sigaction_signal(SIGCHLD, remove_zombies);
+    // ignore sigchld, but set it anyway such that we don't
+    // receive anything on on stopped children (SA_NOCLDSTOP)
```
on on

blarz reviewed (changes requested):
Some small typos, otherwise looks good!

Also: there are some autostart related tests in test_herbstluftwm.py, maybe they can be removed or update?

blarz commented:
> Some small typos, otherwise looks good!
> 
> Also: there are some autostart related tests in test_herbstluftwm.py, maybe they can be removed or update?

At least the `# TODO: verify the path as soon as we have an autostart object` comment seems like something that could be done (maybe in a different PR?)

t-wissmann commented:
I've fixed the typos and moved the test cases to `test_autostart.py`. The comment in `test_autostart_path` however requires another attribute that does not exist yet, so I think this should be done in separate PR.

blarz commented:
> I've fixed the typos and moved the test cases to `test_autostart.py`. The comment in `test_autostart_path` however requires another attribute that does not exist yet, so I think this should be done in separate PR.

Thanks for the fixes. And makes sense :)
